src/main/java/com/hbm/entity/projectile/EntityMeteor.java

```diff
@@ -37,23 +114,57 @@ public void onUpdate() {
 		this.motionY -= 0.03;
 		if(motionY < -2.5)
 			motionY = -2.5;
-		
+
 		this.moveEntity(motionX, motionY, motionZ);
 
-		if(!this.worldObj.isRemote && this.onGround && this.posY < 260) {
-			
-			worldObj.createExplosion(this, this.posX, this.posY, this.posZ, 5 + rand.nextFloat(), !safe);
-			if(WorldConfig.enableMeteorTails) {
-				ExplosionLarge.spawnParticles(worldObj, posX, posY + 5, posZ, 75);
-				ExplosionLarge.spawnParticles(worldObj, posX + 5, posY, posZ, 75);
-				ExplosionLarge.spawnParticles(worldObj, posX - 5, posY, posZ, 75);
-				ExplosionLarge.spawnParticles(worldObj, posX, posY, posZ + 5, 75);
-				ExplosionLarge.spawnParticles(worldObj, posX, posY, posZ - 5, 75);
+		if(!this.worldObj.isRemote && this.posY < 260) {
+			clearMeteorPath(worldObj, (int)this.posX, (int)this.posY, (int)this.posZ);
+			if(this.onGround) {
+				worldObj.createExplosion(this, this.posX, this.posY, this.posZ, 5 + rand.nextFloat(), !safe);
+
+				if(WorldConfig.enableMeteorTails) {
+					ExplosionLarge.spawnRubble(worldObj, this.posX, this.posY, this.posZ, 25);
+
+					ExplosionLarge.spawnParticles(worldObj, posX, posY + 5, posZ, 75);
+					ExplosionLarge.spawnParticles(worldObj, posX + 5, posY, posZ, 75);
+					ExplosionLarge.spawnParticles(worldObj, posX - 5, posY, posZ, 75);
+					ExplosionLarge.spawnParticles(worldObj, posX, posY, posZ + 5, 75);
+					ExplosionLarge.spawnParticles(worldObj, posX, posY, posZ - 5, 75);
+				}
+
+				// Bury the meteor into the ground
+				int spawnPosX = (int) (Math.round(this.posX - 0.5D) + (safe ? 0 : (this.motionZ * 5)));
+				int spawnPosY = (int) Math.round(this.posY - (safe ? 0 : 4));
+				int spawnPosZ = (int) (Math.round(this.posZ - 0.5D) + (safe ? 0 : (this.motionZ * 5)));
+
+				(new Meteorite()).generate(worldObj, rand, spawnPosX, spawnPosY, spawnPosZ, safe, true, true);
+				clearMeteorPath(worldObj, spawnPosX, spawnPosY, spawnPosZ);
+
+				// Sound
+				if(this.audioFly != null) this.audioFly.stopSound();

```

Comment:
considering `audioFly` is only initialized when `worldObj.isRemote`, i don't see how this would ever run